lib/ruby_smb/server/server_client/negotiation.rb

```diff
@@ -0,0 +1,149 @@
+require 'securerandom'
+
+module RubySMB
+  class Server
+    class ServerClient
+      module Negotiation
+        #
+        # Handle an SMB negotiation request. Once negotiation is complete, the state will be updated to :session_setup.
+        # At this point the @dialect will have been set along with other dialect-specific values.
+        #
+        # @param [String] raw_request the negotiation request to process
+        def handle_negotiate(raw_request)
+          response = nil
+          case raw_request[0...4]
+          when "\xff\x53\x4d\x42".b
+            request = SMB1::Packet::NegotiateRequest.read(raw_request)
+            response = do_negotiate_smb1(request)
+          when "\xfe\x53\x4d\x42".b
+            request = SMB2::Packet::NegotiateRequest.read(raw_request)
+            response = do_negotiate_smb2(request)
+          end
+
+          if response.nil?
+            disconnect!
+          else
+            send_packet(response)
+          end
+
+          nil
+        end
+
+        def do_negotiate_smb1(request)
+          if request.dialects.map(&:dialect_string).include?(Client::SMB1_DIALECT_SMB2_WILDCARD)
+            response = SMB2::Packet::NegotiateResponse.new
+            response.smb2_header.credits = 1
+            response.security_mode.signing_enabled = 1
+            response.dialect_revision = 0x02ff
+            response.server_guid = @server.server_guid
+
+            response.max_transact_size = 0x800000
+            response.max_read_size = 0x800000
+            response.max_write_size = 0x800000
```

Comment:
Probably not right now because this is the only place where they're used. Once they need to be used elsewhere, then it would probably make sense to put them in constants or even as an instance variable if any of the values need to be client specific. Another option could be to set them using the `initial_value` of the `SMB2::Packet::NegotiateResponse` class if they're going to be server-specific and not unique to particular client connections.